### PR TITLE
autorandr: 855c18b -> 1.1

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Update package and also fix `predetect` script usage

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/misc/autorandr.nix [hunk 1]
```diff
@@ -22,22 +22,16 @@ in {
...
-      serviceConfig = {
-        ExecStart = "${pkgs.autorandr}/bin/autorandr --batch --change --default default";
-        Type = "oneshot";
-        RemainAfterExit = false;
-      };
       wantedBy = [ "sleep.target" ];
     };
 
+    services.udev.extraRules = ''
+      ACTION=="change", SUBSYSTEM=="drm", RUN+="${pkgs.bash}/bin/sh -c '${pkgs.systemd}/bin/systemctl is-active -q graphical.target && ${pkgs.systemd}/bin/systemctl start autorandr.service'"
```
Comment (Mic92) [inline nixos/modules/services/misc/autorandr.nix]:
```
ACTION=="change", SUBSYSTEM=="drm", TAG+="systemd", ENV{SYSTEMD_WANTS}="autorandr.service"
```

Comment (gnidorah) [inline nixos/modules/services/misc/autorandr.nix]: @Mic92 Great, thank you! Will ask this to go in upstream.

Comment (gnidorah) [inline nixos/modules/services/misc/autorandr.nix]: @Mic92 After a better testing it seems that the rule you suggested doesn't work at monitor replug. Insisting on my rule line then and reopening PR... https://github.com/phillipberndt/autorandr/issues/61#issuecomment-305955669

Comment (Mic92) [inline nixos/modules/services/misc/autorandr.nix]: Then I would rename the upstream unit to `autorandr@.service` and use `BindsTo` of systemd
to make the service restart, whenever the device appears/disappears:

```
ACTION=="change", SUBSYSTEM=="drm", TAG+="systemd", ENV{SYSTEMD_WANTS}="autorandr@%E{DEVNAME}.service"
```

If this does also not work. I would at least remove the raise condition (systemctl is-active check) and make systemctl non-blocking:

 `${pkgs.systemd}/bin/systemctl start --no-block autorandr.service`

Comment (gnidorah) [inline nixos/modules/services/misc/autorandr.nix]: @Mic92 
> set BindsTo to the device node to systemd to make the service restart

Sorry, can't get it. People use different devices. This is why the universal match of `SUBSYSTEM=="drm"` is used.

**Comment by gnidorah** [comment]
@Mic92 Fixed in upstream. Thank you much for the help!

File: nixos/modules/services/misc/autorandr.nix [hunk 2]
```diff
@@ -22,19 +22,9 @@ in {
 
     environment.systemPackages = [ pkgs.autorandr ];
 
-    # systemd.unitPackages = [ pkgs.autorandr ];
+    systemd.packages = [ pkgs.autorandr ];
```
Comment (Mic92) [inline nixos/modules/services/misc/autorandr.nix]: I have enabled the service and it also seems to be trigger:

```
services.autorandr.enable = true;
```

```
Jun 08 12:08:56 turingmachine autorandr[12586]: Failed to load profile 'default': Profile not found (line 1035)
Jun 08 12:08:56 turingmachine autorandr[12586]: Running autorandr as joerg for display :0
Jun 08 12:08:56 turingmachine systemd[1]: Started autorandr execution hook.
```

So the idea would be to make a `default` profile in my user account for autorandr?

Comment (gnidorah) [inline nixos/modules/services/misc/autorandr.nix]: @Mic92 You will need to manually configure autorandr to make use of it. Default is often created as a symlink to desired profile https://github.com/phillipberndt/autorandr#how-to-use

**Comment by gnidorah** [comment]
@Mic92 Thank you!

**Comment by grahamc** [comment]
@gnidorah when trying to use this service I get:

```
Feb 21 10:27:16 Morbo systemd[1]: Starting autorandr execution hook...
Feb 21 10:27:16 Morbo autorandr[26523]: /bin/sh: xrandr: command not found
Feb 21 10:27:16 Morbo autorandr[26523]: Failed to run xrandr (line 439)
Feb 21 10:27:16 Morbo autorandr[26523]: Running autorandr as grahamc for display :0
```

any ideas on what is up?

**Comment by Mic92** [comment]
inpurity ...

**Comment by Mic92** [comment]
There should be a `makeWrapper` in autorandr that adds xrandr to $PATH of autorandr.

**Comment by gnidorah** [comment]
@grahamc @Mic92 https://github.com/NixOS/nixpkgs/pull/35314

BTW autorandr is X11 only thing, no Wayland support. What is the best way to indicate that?
And that means you need `services.xserver.enable = true;` which adds `xorg.xrandr` to `environment.systemPackages`. So the only real use of above PR is autorandr package under non-NixOS distro with missing `xrandr` in path.

**Pull request merged**